edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Core/device/ConnectionInfo.cs

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Hub.Core.Device
+{
+    using System;
+    using System.Threading;
+    using Microsoft.Azure.Devices.Edge.Hub.Core.Identity;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Logging;
+
+    public class ConnectionInfo : IConnectionInfo
+    {
+        public Option<IIdentity> KnownParent { get; private set; }
+
+        public ConnectionInfo(bool isDirectClient)
+        {
+            this.IsDirectClient = isDirectClient;
+        }
+
+        public bool IsDirectClient { get; }
+
+        public void BindToParent(IIdentity parent)
+        {
+            Preconditions.CheckNotNull(parent);
+
+            if (this.KnownParent.Exists(prev => !string.Equals(prev.Id, parent.Id)))
+            {
+                // this should not happen
```

Comment:
So this should be an error condition?

Reply:
As we do not have proper "disconnection" (edgeHub uses an idle timeout to disconnect clients that went silent and without subscriptions), I give a little chance that maybe this can happen in sort-of valid circumstances. E.g. let's say that we have a parent edge and two child edge, and a device that connects to the first child, and subscribes. Then the knowParent will be set to that child edge.
Then this device disconnects from that child edge. Now the parent edge will not know about that, because there is no dedicated notification. Child edge hub will unsubscribe from anything, and the parent edgeHub is going to see that, but other than that nothing happens.
Then this child device connects to the other child edge. I think this is possible as I can set multiple parents to a device. At this moment the parent edgeHub still believes that the device belongs to the previous child edge.
Now the device subscribes to something. In this case the new child edge is going to send a subscription change, and this modification notices that there is a new 'guess' for the parent of the device and the call comes here, changing the parent.

I think it is a valid scenario (and maybe because of that I should remove the comment as "this should not happen" instead.